src/App.tsx

```diff
@@ -1,39 +1,60 @@
 import React from 'react';
 import './App.scss';
+import { Clock } from './components/Clock/Clock';
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+type Props = {};
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
-
-  // this code stops the timer
-  window.clearInterval(timerId);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
-
-        {' time is '}
+interface State {
+  clockName: string,
+  isClockShown: boolean,
+}
 
-        <span className="Clock__time">
-          {today.toUTCString().slice(-12, -4)}
-        </span>
+export class App extends React.Component <Props, State> {
+  state = {
+    clockName: 'Clock-0',
+    isClockShown: true,
+  };
+
+  timerId = 0;
+
+  componentDidMount(): void {
+    window.addEventListener('click', this.toggleClock);
+    window.addEventListener('contextmenu', this.toggleClock);
+
+    // This code starts a timer
+    this.timerId = window.setInterval(() => {
+      this.setState({ clockName: getRandomName() });
+    }, 3300);
+  }
+
+  componentWillUnmount(): void {
+    // this code stops the timer
+    window.clearInterval(this.timerId);
+    window.removeEventListener('click', this.toggleClock);
+    window.removeEventListener('contextmenu', this.toggleClock);
+  }
+
+  toggleClock = (event:Event) => {
+    event.preventDefault();
+    const isShown = this.state.isClockShown;
+
+    this.setState({
+      isClockShown: !isShown,
+    });
+  };
```

Comment:
it doesn't work exactly as intented with this toggle function and always changing the `isClockShown` state to the opposite, because for now you can just keep clicking with either left or right click and the clock will aways toggle. The goal is to change the state to true or false depending on whether right or left mouse button has been clicked. For that you need to create 2 separate functions  with slightly different purpose